src/solver/cache.rs

```diff
@@ -171,14 +171,26 @@ impl<D: DependencyProvider> SolverCache<D> {
         match self.version_set_candidates.get(&version_set_id) {
             Some(candidates) => Ok(candidates),
             None => {
-                let package_name = self.provider.version_set_name(version_set_id);
-                let candidates = self.get_or_cache_candidates(package_name).await?;
+                let package_name_id = self.provider.version_set_name(version_set_id);
+
+                tracing::trace!(
+                    "Getting matching candidates for package: {:?}",
+                    self.provider.display_name(package_name_id).to_string()
+                );
```

Comment:
`display_name` returns something that implements `Display`. You can then use `{}` to format it directly without having to convert to a string first:

```suggestion
                tracing::trace!(
                    "Getting matching candidates for package: {}",
                    self.provider.display_name(package_name_id)
                );
```